packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -74,46 +73,55 @@ contract L2Genesis is Script, Artifacts {
         0xa0Ee7A142d267C1f36714E4a8F75612F20a79720
     ];
 
-    string internal outfile;
-
-    /// @dev Reads the deploy config, sets `outfile` which is where the `vm.dumpState` will be saved to, and
-    ///      loads in the addresses for the L1 contract deployments.
-    function setUp() public override {
-        Artifacts.setUp();
-
-        string memory path = string.concat(vm.projectRoot(), "/deploy-config/", deploymentContext, ".json");
-        vm.etch(address(cfg), vm.getDeployedCode("DeployConfig.s.sol:DeployConfig"));
-        vm.label(address(cfg), "DeployConfig");
-        vm.allowCheatcodes(address(cfg));
-        cfg.read(path);
-
-        outfile = string.concat(vm.projectRoot(), "/deployments/", deploymentContext, "/genesis-l2.json");
+    function name() public pure override returns (string memory) {
+        return "L2Genesis";
+    }
 
-        _loadAddresses(string.concat(vm.projectRoot(), "/deployments/", deploymentContext, "/.deploy"));
+    function artifactDependencies() internal view returns (L1Dependencies memory l1Dependencies_) {
+        console.log("retrieving L1 deployments from artifacts");
+        return L1Dependencies({
+            deployedL1CrossDomainMessengerProxy: mustGetAddress("L1CrossDomainMessengerProxy"),
+            deployedL1StandardBridgeProxy: mustGetAddress("L1StandardBridgeProxy"),
+            deployedL1ERC721BridgeProxy: mustGetAddress("L1ERC721BridgeProxy")
+        });
     }
 
     /// @dev Sets the precompiles, proxies, and the implementation accounts to be `vm.dumpState`
     ///      to generate a L2 genesis alloc.
     /// @notice The alloc object is sorted numerically by address.
-    function run() public {
-        _dealEthToPrecompiles();
-        _setPredeployProxies();
-        _setPredeployImplementations();
+    function runWithStateDump() public {
+        runWithOptions(OutputMode.DEFAULT_LATEST, artifactDependencies());
+    }
 
+    // @dev This is used by op-e2e to have a version of the L2 allocs for each upgrade.
+    function runWithAllUpgrades() public {
+        runWithOptions(OutputMode.OUTPUT_ALL, artifactDependencies());
+    }
+
+    function runWithOptions(OutputMode _mode, L1Dependencies memory _l1Dependencies) public {
+        dealEthToPrecompiles();
+        setPredeployProxies();
+        setPredeployImplementations(_l1Dependencies);
```

Comment:
This creates a really good way for us to standardize on the initial release of bedrock inside of `setPredeployImplementations`, we can copy/paste the initial release bytecode to be used there. This would mean we want a "development" output mode which uses the contracts based on the source code when the command runs. No need to do this right now but curious what you think of this approach?

Reply:
Ah this dev mode is local delta